tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CreatePurchasingDocumentServiceTests.cs

```diff
@@ -13,32 +14,63 @@ namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Services
     [Parallelizable(ParallelScope.All)]
     internal sealed class CreatePurchasingDocumentServiceTests
     {
+        [Test]
+        public void Constructor_NullParameter_ThrowsArgumentNullException()
+        {
+            var builder = CreatePurchasingDocumentServiceBuilder
+                .Create()
+                .WithPatientNumbersCsvWriter(null);
+
+            Assert.Throws<ArgumentNullException>(() => builder.Build());
+        }
+
         [TestCase(true, false)]
         [TestCase(false, true)]
         [TestCase(false, false)]
-        public void CreateDocumentAsync_ArgumentsAreNull_ReturnNullArgumentException(bool hasStream, bool hasOrder)
+        public void CreatePatientNumbersCsvAsync_ArgumentsAreNull_ReturnNullArgumentException(bool hasStream, bool hasOrder)
         {
-            var purchasingDocumentService = new CreatePurchasingDocumentService();
+            var context = CreatePurchasingDocumentServiceTestContext.Setup();
 
             Assert.ThrowsAsync<ArgumentNullException>(() =>
-                purchasingDocumentService.CreateDocumentAsync(
+                context.CreatePurchasingDocumentService.CreatePatientNumbersCsvAsync(
                     hasStream ? new MemoryStream() : null,
                     hasOrder ? OrderBuilder.Create().Build() : null));
         }
 
         [Test]
-        public async Task CreateDocumentAsync_DocumentIsCreated_StreamIsAsExpected()
+        public async Task CreatePatientNumbersCsvAsync_DocumentIsCreated_WriteRecordsAsyncCalledOnce()
         {
-            var purchasingDocumentService = new CreatePurchasingDocumentService();
-
+            var context = CreatePurchasingDocumentServiceTestContext.Setup();
             await using var stream = new MemoryStream();
             var order = OrderBuilder.Create().Build();
-            await purchasingDocumentService.CreateDocumentAsync(stream, OrderBuilder.Create().Build());
 
-            var expectedStreamContent = stream.ToArray();
-            var actualStreamContent = Encoding.UTF8.GetBytes($"Call off Party Id{Environment.NewLine}{order.OrderId}{Environment.NewLine}");
+            await context.CreatePurchasingDocumentService.CreatePatientNumbersCsvAsync(stream, order);
 
-            expectedStreamContent.Should().BeEquivalentTo(actualStreamContent);
+            context.PatientCsvWriterMock.Verify(
+                x => x.WriteRecordsAsync(It.IsAny<Stream>(), It.IsAny<IEnumerable<PatientNumbersPriceType>>()),
```

Comment:
Would it be possible to validate the mapping part of the `CreatePatientNumbersCsvAsync` method? So make sure the list of `PatientNumbersPriceType` is what you expect based on the supplied order.